### PR TITLE
Changed the event priorities

### PR DESCRIPTION
**Pull request by Nyholm**
This is the registered events in standard symfony.

Listener Class Name | Priority
---------------------------- | ---------
ProfilerListener | 1024
TestSessionListener | 192
SessionListener | 128
RouterListener | 32
LocaleListener | 16
Firewall | 8

([source](http://symfony.com/doc/current/reference/dic_tags.html#kernel-request))

Current behavior is to set the application and transaction name after the firewall. This will lead to that transactions interrupted by the firewall are not getting the proper application key. They will be registered with the license key defined in php.ini. (See #87)

This PR splits the `RequestListener::onCoreRequest` into three functions. One that names the application, one that ignores the transaction and one that names the transaction. The one that set the application name should have as high priority as possible (255 is a nice number) because some listeners might stop the event propagation and return a response. 

Since we want to ignore some routes we should call the `RequestListener::setIgnoreTransaction` as soon as the RouterListener has executed. That's why we should use priority 31.

For the transaction name: If we have too high priority here we might not have sufficient data for the `TransactionNamingStrategyInterface`. If we have to too low, say after the firewall, some transaction will slip through with no name. IMHO, we should make sure we have all data possible. When an exception happens in one of the listeners is it okey not to have a proper name.

This PR will solve issue #87.

Update 2015-05-15 14:17: I've rewrote the description because of the changes that @Lumbendil suggested.

### CODE REVIEW

File: Listener/RequestListener.php [hunk 1]
```diff
@@ -78,7 +68,47 @@ public function onCoreRequest(GetResponseEvent $event)
...
     }
+
+    /**
+     * Make sure that it is not a subrequest and that we ignore paths that should be ignored
+     *
+     * @param GetResponseEvent $event
+     *
+     * @return \Symfony\Component\HttpFoundation\Request|null Return a Request or null iff it is not a master request
+     */
+    protected function validateRequest(GetResponseEvent $event)
```
Comment (rande) [inline Listener/RequestListener.php]: This should return true or false, and the argument must be the request object itself.

**Comment by rande** [comment]
@odolbeau can you review this PR ?

File: Listener/RequestListener.php (hunk 1 above)
Comment (Nyholm) [inline Listener/RequestListener.php]: Yes, you are correct.

**Comment by Lumbendil** [comment]
@rande @Nyholm atm some 404 (triggered by the router) are being shown with "PHP Application". Would it be possible to change the name and afterwards tell it not to track that request?

**Comment by Nyholm** [comment]
@Lumbendil You are right.. That should be possible. I'll update the PR.

**Comment by Nyholm** [comment]
The PR is updated. The `RequestListener::onCoreRequest` has been replaced by the following functions: 
* `RequestListener::setApplicationName`
* `RequestListener::setIgnoreTransaction`
* `RequestListener::setTransactionName`

**Comment by rande** [comment]
@Nyholm Can you add documentation about the current request flow ? It will help people to understand how that work and related issues.

**Comment by Nyholm** [comment]
1) A request comes in and the first thing we do is to `setApplicaitonName` so that we use the correct license key and name. 

2) The `RouterListener` might throw a 404 or add routing values to the request.

3) If no 404 was thrown we `setIgnoreTransaction` which means that we call `$this->interactor->ignoreTransaction();` if we have configured to ignore the route. 

4) The Firewall is the next interesting thing that will happen. It could change the controller or throw a 403. 

5) The developer might have configured many more request listeners that will now execute and possibly add stuff to the request.

6) We will execute `setTransactionName` to use our `TransactionNamingStrategyInterface` to set a nice name. 

-------

All 6 steps will be executed for a normal request. Exceptions to this is 404 and 403 responses that will be created in step 2 and step 4 respectively. If an exception to these step occurs (I'm not talking about `\Exception`) you will have the transaction logged with the correct license key but you do not have the proper transaction name. (Because of the reasons in the PR description.)

**Comment by rande** [comment]
@Nyholm Can you put this note into the README.md file ;)

**Comment by Nyholm** [comment]
Oh, sorry. Sure. I've updated the PR now.

File: README.md [hunk 2]
```diff
@@ -186,6 +186,20 @@ The bundle provide a [Capifony](http://capifony.org) recipe to automate the depl
 
 It makes one request per `app_name`, due roll-up names are not supported by Data REST API.
 
+## The flow of the Request
+
+1. A request comes in and the first thing we do is to `setApplicaitonName` so that we use the correct license key and name.
```
Comment (rande) [inline README.md]: ``setApplication`` there is a typo

Comment (Nyholm) [inline README.md]: Thank you

File: README.md [hunk 3]
```diff
@@ -186,6 +186,20 @@ The bundle provide a [Capifony](http://capifony.org) recipe to automate the depl
 
 It makes one request per `app_name`, due roll-up names are not supported by Data REST API.
 
+## The flow of the Request
```
Comment (Lumbendil) [inline README.md]: I think "Flow of the Request" would read better :)

**Comment by rande** [comment]
@Nyholm I want @odolbeau to review this PR, this month is almost off in France, so you might need to wait a bit before this PR can get merge.

**Comment by Nyholm** [comment]
Thank you for reviewing this so fast and giving me comments. 

Yeah, let's wait for @odolbeau. I want to make sure we haven't missed anything.

**Comment by odolbeau** [comment]
:+1: 
Thanks for the very clear description! :)

**Comment by rande** [comment]
@odolbeau so you can merge the request ;)

**Comment by odolbeau** [comment]
Merged!
Thanks @Nyholm! :)

**Comment by rande** [comment]
@odolbeau GG, Thanks @Nyholm ;)

**Comment by Nyholm** [comment]
Thank you for such quick feedback and thank you for merging.

**Pull request merged**